app/controllers/merchant/dashboard_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Merchant::DashboardController < ApplicationController
+
+  before_action :require_merchant
+
+  def require_merchant
+    render file: "/public/404" unless current_merchant?
+  end
+
+  def index
+
+  end
```

Comment:
Potential for small refactor here to get this on one line.

Reply:
We've done one-line approach in other controllers.

Reply:
will do!